sycl/include/CL/sycl/half_type.hpp

```diff
@@ -74,13 +74,21 @@ class __SYCL_EXPORT half {
     return ret;
   }
 
+  // Operator neg
+  constexpr half &operator-() {
+    Buf ^= 0x8000;
+    return *this;
+  }
+
   // Operator float
   operator float() const;
 
   template <typename Key> friend struct std::hash;
 
 private:
+  constexpr explicit half(uint16_t x) : Buf(x) {}
   uint16_t Buf;
+  friend std::numeric_limits<class cl::sycl::detail::half_impl::half>;
```

Comment:
I didn't realize all is fwd-declared and therefore
```suggestion
  friend std::numeric_limits<half_impl::half>;
```
or
```suggestion
  friend std::numeric_limits<sycl::half>;
```
works too. Do we prefer one of these shorter options?

Reply:
I have no preference.